src/pages/checkout/index.tsx

```diff
@@ -46,6 +63,25 @@ const Checkout: React.FC = () => {
 	const [comment, setComment] = React.useState<string>('');
 	const [popupText, setPopupText] = useState('');
 	const [isAgreed, setIsAgreed] = useState(false);
+	const [promocodeError, setPromocodeError] = useState('');
+	const [inputPromoValue, setInputPromoValue] = useState('');
+	const discountInitial = {
+		code: '',
+		conditions: '',
+		discount: 0,
+		discount_amount: 0,
+		end_time: null,
+		id: 0,
+		image: null,
+		is_active: false,
+		is_constant: false,
+		name: '',
+		new_total_price: 0,
+		promotion_type: '',
+		slug: '',
+		start_time: '',
+	};
+	const [discount, setDiscount] = useState<Promo>(discountInitial);
```

Comment:
А здесь вот так соответственно:

```suggestion
	const [discount, setDiscount] = useState<Promo | null>(null);
```